snapcraft/internal/cache/_snap.py

```diff
@@ -76,8 +76,24 @@ def prune(self, *, keep_revision):
                         'Unable to purge snap {}.'.format(cached_snap))
         return pruned_files_list
 
+    def get(self, *, snap_name, revision=None):
+        cached_snaps = []
+        for f in os.listdir(self.snap_cache_dir):
+            if f.startswith(snap_name):
```

Comment:
isn't this risky with snaps whose name is a subset of other snaps? Like hello and hello-world?